programs/drift_vaults/src/state/vault_depositor.rs

```diff
@@ -510,6 +525,57 @@ impl VaultDepositor {
 
         Ok(user_withdraw_amount)
     }
+
+    pub fn realize_profits(
+        self: &mut VaultDepositor,
+        vault_equity: u64,
+        vault: &mut Vault,
+        now: i64,
+    ) -> Result<u64> {
+        let management_fee = vault.apply_management_fee(vault_equity, now)?;
+
+        let vault_shares_before = self.checked_vault_shares(vault)?;
+        let total_vault_shares_before = vault.total_shares;
+        let user_vault_shares_before = vault.user_shares;
+
+        let n_shares = self.vault_shares;
+        let amount: u64 =
+            depositor_shares_to_vault_amount(n_shares, vault.total_shares, vault_equity)?;
+
+        let total_amount =
+            depositor_shares_to_vault_amount(self.vault_shares, vault.total_shares, vault_equity)?;
+
+        let profit_share: u64 = self
+            .calculate_profit_share_and_update(amount, total_amount, vault)?
+            .cast()?;
+
+        let profit_share_shares: u128 =
+            vault_amount_to_depositor_shares(profit_share, vault.total_shares, vault_equity)?;
+
+        self.decrease_vault_shares(profit_share_shares, vault)?;
```

Comment:
should this affect net deposits? it does in withdraw()

Reply:
i've updated the way the profit share is deducted, so the user will have to know upon request their would-be balance